mlflow/xgboost/__init__.py

```diff
@@ -705,6 +706,15 @@ def infer_model_signature(input_example):
 
         return model
 
+    def train(_log_models, original, *args, **kwargs):
+        current_sklearn_session = _SklearnTrainingSession.get_current_session()
+        if current_sklearn_session is None or (
+            current_sklearn_session is not None and current_sklearn_session.should_log()
```

Comment:
```suggestion
            current_sklearn_session.should_log()
```

Do we need `current_sklearn_session is not None and`?

Reply:
yes, otherwise it might call `None.should_log`

what about change `should_log` to be a staticmethod ? then we can directly use `if _SklearnTrainingSession.should_log` here